zebra-network/src/peer_set/candidate_set.rs

```diff
@@ -178,20 +186,27 @@ where
     ///
     /// See [`update_initial`][Self::update_initial] for details.
     async fn update_timeout(&mut self, fanout_limit: Option<usize>) -> Result<(), BoxError> {
-        // CORRECTNESS
+        // SECURITY
         //
-        // Use a timeout to avoid deadlocks when there are no connected
-        // peers, and:
-        // - we're waiting on a handshake to complete so there are peers, or
-        // - another task that handles or adds peers is waiting on this task
-        //   to complete.
-        if let Ok(fanout_result) =
-            timeout(constants::REQUEST_TIMEOUT, self.update_fanout(fanout_limit)).await
-        {
-            fanout_result?;
-        } else {
-            // update must only return an error for permanent failures
-            info!("timeout waiting for the peer service to become ready");
+        // Rate limit sending `GetAddr` messages to random peers.
```

Comment:
Just "peers" is enough 😄 
```suggestion
        // Rate limit sending `GetAddr` messages to peers.
```